src/overlays/actors/ovl_Obj_Ghaka/z_obj_ghaka.c

```diff
@@ -21,26 +34,160 @@ const ActorInit Obj_Ghaka_InitVars = {
     (ActorFunc)ObjGhaka_Update,
     (ActorFunc)ObjGhaka_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Ghaka_0x80B3C260/func_80B3C260.asm")
+static Vec3f D_80B3C960 = { 0.0f, 0.0f, 0.0f };
+
+static InitChainEntry D_80B3C96C[] = {
+    ICHAIN_U8(targetMode, 0, ICHAIN_CONTINUE),
+    ICHAIN_F32(targetArrowOffset, 30, ICHAIN_STOP),
+};
+
+extern Gfx D_06001A20[];
+extern Gfx D_06001980[];
+
+void func_80B3C260(ObjGhaka* this) {
+    if (gSaveContext.weekEventReg[20] & 0x20) {
+        this->dyna.actor.world.pos.z = this->dyna.actor.home.pos.z + 100.0f;
+    }
+    this->actionFunc = func_80B3C39C;
+}
+
+void func_80B3C29C(ObjGhaka* this) {
+    this->actionFunc = func_80B3C4E0;
+}
+
+void func_80B3C2B0(ObjGhaka* this) {
+    this->actionFunc = func_80B3C624;
+}
+
+void func_80B3C2C4(ObjGhaka* this, GlobalContext* globalCtx) {
+    if ((gSaveContext.weekEventReg[20] & 0x20) == 0) {
+        Actor_SpawnWithParentAndCutscene(&globalCtx->actorCtx, globalCtx, ACTOR_BG_GORON_OYU, 0.0f, 25.0f, 261.0f, 0, 0,
+                                         0, 0, this->dyna.actor.cutscene, this->dyna.actor.unk20, 0);
+    } else {
+        Actor_SpawnWithParentAndCutscene(&globalCtx->actorCtx, globalCtx, ACTOR_BG_GORON_OYU, 0.0f, 25.0f, 261.0f, 0, 0,
+                                         0, 1, -1, this->dyna.actor.unk20, 0);
+    }
+}
+
+void func_80B3C39C(ObjGhaka* this, GlobalContext* globalCtx) {
+    ActorPlayer* player = PLAYER;
+    s16 distDiff;
+
+    distDiff = this->dyna.actor.yawTowardsPlayer - this->dyna.actor.shape.rot.y;
+    if (func_800B84D0(&this->dyna.actor, globalCtx)) {
+        func_80B3C29C(this);
+    } else if (this->dyna.actor.xzDistToPlayer < 100.0f || this->dyna.actor.isTargeted) {
+        if (distDiff <= -0x5556 || distDiff >= 0x5556) {
+            func_800B863C(&this->dyna.actor, globalCtx);
+            if (player->unk14B == 1) {
+                this->dyna.actor.textId = 0xCF3;
+            } else {
+                this->dyna.actor.textId = 0xCF2;
+            }
+        }
+    }
+    if (this->dyna.unk148 < 0.0f && !(gSaveContext.weekEventReg[20] & 0x20) && player->unk14B == 1) {
+        func_80B3C2B0(this);
+    } else {
+        player->unkA70 &= ~0x10;
+        this->dyna.unk148 = 0.0f;
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Ghaka_0x80B3C260/func_80B3C29C.asm")
+void func_80B3C4E0(ObjGhaka* this, GlobalContext* globalCtx) {
+    s32 temp_v0;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Ghaka_0x80B3C260/func_80B3C2B0.asm")
+    temp_v0 = func_80152498(&globalCtx->msgCtx);
+    if (temp_v0 == 5) {
```

Comment:
If you change `else` `else if` with temp_v0 be a switch to a switch, does it let you remove the '& 0xFFFFFFFFu'

Reply:
I did try that and it didn't work :/